### PR TITLE
feat: ✨ Allow syn-divider as a separator in meta-navigation of syn-header

### LINKED ISSUE #1130: feat: ✨ divider between items in meta-navigation of syn-header
# Checklist (for the issue creator)

- [ ] I have looked at the documentation site and was not able to find the required feature.
- [ ] I have looked into the Figma boards and was not able to find the required feature.
- [ ] I would be willing to provide a PR for this feature.

# Component

I suggest to add the feature to the following component: syn-header

# Story

As a developer, I would like to add a vertical divider in slots of the meta-navigation of syn-header so that existing functional icons can be visually grouped and separated, improving clarity and alignment with design requirements.

# Acceptance criteria

- [ ] All DoD items successfully completed
- [ ] set syn-divider to 100% height if slotted in meta-navigation
- [ ] create a story how to use a vertical divider in the slots of the meta-navigation `<syn-divider vertical=""></syn-divider>`

```
<syn-header label="App Name">
  <nav slot="meta-navigation">
    <syn-icon-button name="apps" label="Apps"></syn-icon-button>
    <syn-divider vertical=""></syn-divider>
    <syn-icon-button name="account_circle" label="Account">
</syn-icon-button>
```

### PR DESCRIPTION
**Pull request by schilchSICKAG**
# Pull Request

## 📖 Description

This PR adds the possiblity to use `<syn-divider>` as a separator between items in the `<syn-header>` metanavigation.
It will automatically set the `height`, as well as add the `vertical` property to directly slotted dividers.

You may use it via:

```html
<syn-header label="App Name">
  <nav slot="meta-navigation">
    <syn-icon-button name="settings_outline" label="Settings"></syn-icon-button>
    <syn-icon-button name="insert_chart_outlined" label="Analytics"></syn-icon-button>
    <syn-divider></syn-divider>
    <syn-icon-button name="dark_mode" label="Dark Mode"></syn-icon-button>
    <syn-divider></syn-divider>
    <syn-icon-button name="language" label="Language"></syn-icon-button>
    <syn-icon-button name="login" label="Login"></syn-icon-button>
  </nav>
</syn-header>
```

or via directly slotting the `<syn-divider>` via

```html
<syn-header label="App Name">
  <syn-icon-button slot="meta-navigation" name="settings_outline" label="Settings"></syn-icon-button>
  <syn-icon-button slot="meta-navigation" name="insert_chart_outlined" label="Analytics"></syn-icon-button>
  <syn-divider slot="meta-navigation"></syn-divider>
  <syn-icon-button slot="meta-navigation" name="dark_mode" label="Dark Mode"></syn-icon-button>
  <syn-divider slot="meta-navigation"></syn-divider>
  <syn-icon-button slot="meta-navigation" name="language" label="Language"></syn-icon-button>
  <syn-icon-button slot="meta-navigation" name="login" label="Login"></syn-icon-button>
</syn-header>
```

`<syn-header>` now also exposes a new `cssproperty` `--metanavigation-item-size` that can be used to define the size of rendered `<syn-icon-buttons>`, as well as the height of `<syn-divider>`

### 🎫 Issues

Closes #1130 

## 👩‍💻 Reviewer Notes

It is NOT possible to use a simple `::slotted(syn-divider)` here because this would only target **directly** slotted elements. When you use an intermediate element to wrap everything up (e.g. a `<nav slot="meta-navigation">`, then the `::slotted` selector is not able to set the style for the inner children. Because of this, I went with a `slotchange` event which seems to work fine. However, this means we are using inline overrides in CSS to achieve the correct layout.

To make it work, I also had to get the size that should be used. To make it easier to sync the value with the size of rendered `syn-icon-button` elements, I exchanged the fixed `font-size` of this elements with a new variable `--metanavigation-item-size` that maps to the old value.

Spacings of the slotted `<syn-divider>` was also altered to make it look the same as in Figma. This is set via inline style, too.

For applying the styles, I opted to set them via `Element.cssText`. This batches the three settings that need to be done into one call and is therefore faster than using three separate `Element.style.setProperty`.

I also made sure to only target directly rendered `<syn-dividers>`. Otherwise, you would not be able to render nested `<syn-dropdown>` items that render `<syn-dividers>` themself.

## 📑 Test Plan

- As usual, check the stories and code.
- I also created a more expressive demo with edge cases that you may use in the `header.stories`. The adjusted story is located at https://github.com/synergy-design-system/synergy-design-system/issues/1130#issuecomment-3989830237.

## ✅ DoD

- [x] I have tested my changes manually.
- [x] I have added automatic tests for my changes (unit- and visual regression tests).
- [x] I have updated the project documentation to reflect my changes (e.g. CHANGELOG, Storybook, ...).
- [x] I have made sure to follow the projects coding and contribution guides.
- [x] I have made sure that the bundle size has changed appropriately.
- [x] I have validated that there are no accessibility errors.
- [x] I have added a changeset, describing my changes (e.g. via `pnpm release.create`).
- [x] I have used design tokens instead of fix css values

### CODE REVIEW

File: packages/components/src/components/header/header.component.ts [hunk 1]
```diff
@@ -128,6 +131,43 @@ export default class SynHeader extends SynergyElement {
...
+    // Set the correct styles for the found dividers
+    // - --spacing: var(--syn-spacing-x-small) makes the spacing slightly narrower,
+    // - align-self: center makes sure the divider is centered in the meta navigation,
+    // - height: var(--metanavigation-item-size) makes sure the divider has the same height as the other items in the meta navigation.
+    // We use this approach as it is faster than 3 separate style changes and we can be sure to not override any other styles that might be set on the divider.
+    const cssText = '--spacing: var(--syn-spacing-x-small); align-self: center; height: var(--metanavigation-item-size);';
+
+    foundDividers
+      .forEach(divider => {
+        divider.setAttribute('vertical', '');
```
[the remark this answers is not in the document]
Comment (schilchSICKAG) [reply packages/components/src/components/header/header.component.ts]: I dont set them all as vertical. I think what I do is pretty sensible: I just search for directly slotted dividers and those who are exactly one level deeper, so this should be applied when using structures like:

<syn-header>
<nav slot="meta-navigation">
  <divider>
</nav>
</syn-header>

or 

<syn-header>
  <divider slot="meta-navigation">
</syn-divider>

For example, when adding a dropdown to the header it will not force the visibility.

File: packages/components/src/components/header/header.component.ts [hunk 2]
```diff
@@ -128,6 +131,43 @@ export default class SynHeader extends SynergyElement {
...
+   * .header__meta-navigation ::slotted(*) { will ONLY work for directly slotted children, but not for children of slotted elements, which is why we need to do this via JavaScript.
+   */
+  private updateMetaNavigation() {
+    const foundDividers: SynDivider[] = [];
+    this.metaNavigationSlot
+      // Get all slotted children.
+      // Needed because we may also have dividers in a wrapper that is used for the meta navigation
+      .assignedElements({ flatten: true })
+      .forEach(el => {
+        if (el.tagName.toLowerCase() === 'syn-divider') {
```
[the remark this answers is not in the document]
Comment (schilchSICKAG) [reply packages/components/src/components/header/header.component.ts]: I seem to have remove the `display: flex` by accident. Added it back now.
